lib/samson/hooks.rb

```diff
@@ -80,6 +124,16 @@ def hooks(name)
 
 Dir["plugins/*/lib"].each { |f| $LOAD_PATH << f } # treat included plugins like gems
 
-Samson::Hooks.plugins.
-  each(&:require).
-  each(&:add_migrations)
+module LoadDecorators
+  def inherited(subclass)
+    Samson::Hooks.fire(:model_defined, subclass)
+    super
+  end
+end
+
+Samson::Hooks.callback :model_defined do |model_class|
```

Comment:
instead of iterating over all plugins every time and doing a bunch of file.exist checks, maybe only pick the plugins that opted in via

```
Samson::Hooks.callback :model_defines, "Stage" -> # load plugin/app/models/stage.rb
```